doc/config/ubuntu-16.04-kinetic.sh

```diff
@@ -20,3 +19,4 @@ export ROS_PACKAGE_PATH=$INSTALL_HPP_DIR/share:$ROS_PACKAGE_PATH
 if [ -f "${INSTALL_HPP_DIR}/etc/hpp-tools/bashrc" ]; then
     source "${INSTALL_HPP_DIR}/etc/hpp-tools/bashrc"
 fi
+export CMAKE_PREFIX_PATH=/opt/openrobots:$CMAKE_PREFIX_PATH
```

Comment:
Is it safe to have it this way ? Is there a problem with this:
```
export CMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH:/opt/openrobots
```

The thing I care about is CMake finding something in robotpkg before the HPP devel space.

Reply:
I reply here to this:

> 
> As ROS seems to overwrite it in devel-premade:16.04:
> 
> > $ echo $CMAKE_PREFIX_PATH
> >
> > $ grep CMAKE_PREFIX_PATH /workspace/config.sh
> > export CMAKE_PREFIX_PATH=/opt/openrobots
> > $ source /workspace/config.sh
> > $ echo $CMAKE_PREFIX_PATH
> > /workspace/install:/opt/ros/kinetic
> 
> which leads to:
> > Could NOT find qpOASES (missing: qpOASES_LIBRARY qpOASES_INCLUDE_DIR)

The CMAKE_PREFIX_PATH is overwritten by ROS setup.sh. This file uses _setup_util.py, which contains:
```py
def _parse_arguments(args=None):
    parser = argparse.ArgumentParser(description='Generates code blocks for the setup.SHELL script.')
    parser.add_argument('--extend', action='store_true', help='Skip unsetting previous environment variables to extend context')
    return parser.parse_known_args(args=args)[0]
```
I don't know whether this would solve the issue. I think I tried some day but I can't remember the result.

Reply:
I don't feel like trying to understand how silly ROS is, we just need for this to work.
I agree that `export CMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH:/opt/openrobots` could prevent issues, so I will try it.

Reply:
> I don't feel like trying to understand how silly ROS is

That's reasonable.

Reply:
If I'm not mistaken, in the docker image LD_LIBRARY_PATH look in openrobots before HPP devel (I had an error related to this). If you change the order of the CMAKE_PREFIX_PATH you should check that both path have the same order.

Reply:
https://github.com/humanoid-path-planner/hpp-doc/blob/b1026cc5a4dac5d2b18e34c2808de4b6c75b33b1/doc/config/ubuntu-16.04-kinetic.sh#L9 nope, this one seems right.

Reply:
That's before ROS, you may want to double check after.

Reply:
Oh, no… ROS would not mess with my environment variables, would it ?
```
root@1844764b8264:/# echo $LD_LIBRARY_PATH

root@1844764b8264:/# source /workspace/config.sh 
root@1844764b8264:/# echo $LD_LIBRARY_PATH
/workspace/install/lib:/workspace/install/lib64:/opt/ros/kinetic/lib:/opt/openrobots/lib:
```
See ? :)

Reply:
It worked: https://gepgitlab.laas.fr/gsaurel/hpp-doc/pipelines/2800